### PR TITLE
[#3784]fix HttpUtils cannot correctly resolve file names

### PR DESCRIPTION
**Pull request by david6969xin**
#3784

### CODE REVIEW

**Comment by liubao68** [comment]
@david6969xin test case not passed

**Comment by david6969xin** [comment]
> @david6969xin test case not passed

@liubao68 ok,i will fix it

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public static String parseParamFromHeaderValue(String headerValue, String paramN
       }
 
       if (paramName.equalsIgnoreCase(trimValue.substring(0, idx))) {
-        return trimValue.replaceAll("\"", "").substring(idx + 1);
+        return trimValue.replaceAll("\"", "").substring(idx + 1).trim();
```
Comment (liubao68) [inline foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java:49]: Is it possible `a=b; a = b `

Comment (liubao68) [reply foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java:49]:
```
    for (String value : headerValue.split(";")) {
      int idx = value.indexOf('=');
      if (idx == -1) {
        continue;
      }

      if (paramName.equalsIgnoreCase(value.substring(0, idx).trim())) {
        return value.substring(idx + 1).trim().replaceAll("\"", "");
      }
    }
```

Is this better?

File: foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/http/TestHttpUtils.java [hunk 2]
```diff
@@ -55,6 +55,21 @@ public void parseParamFromHeaderValue_keyNotFound() {
...
+
+  @Test
+  public void parseParamFromHeaderValue_valueWithBlank() {
+    Assertions.assertEquals("v", HttpUtils.parseParamFromHeaderValue("xx; k=v", "k"));
+  }
+
+  @Test
+  public void parseParamFromHeaderValue_valueWithQuotationMarkAndBlank() {
+    Assertions.assertEquals("v", HttpUtils.parseParamFromHeaderValue("xx;k= \"v\"", "k"));
+  }
```
Comment (liubao68) [inline foundations/foundation-common/src/test/java/org/apache/servicecomb/foundation/common/http/TestHttpUtils.java:71]: Maybe the following cases is enough 

* key1=a;key2=b
* key1=a; key2=b
* key1="a";key2="b"
* key1="a"; key2="b"
* key1="a"; key2="b c.txt"

**Comment by liubao68** [comment]
@david6969xin would you mind to fix the review comments?

**Comment by david6969xin** [comment]
> @david6969xin would you mind to fix the review comments?
yes，i will fix that lately accoding to the review comments @liubao68

File: foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java (hunk 1 above)
Comment (david6969xin) [reply foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/http/HttpUtils.java:49]: sure

**Review by liubao68 — APPROVED** [review]

**Pull request merged**